src/react/hooks/media-query.ts

```diff
@@ -0,0 +1,22 @@
+import {useEffect, useState} from "react";
+
+/**
+ * Custom hook to trigger things on breakpoints
+ */
+export function useMediaQuery (query: string) : boolean
+{
+	const [isActive, setActive] = useState(false);
+
+	useEffect(() =>
+	{
+		const mediaQuery = window.matchMedia(query);
+		setActive(mediaQuery.matches);
+
+		const updater = (event: MediaQueryListEvent) => setActive(event.matches);
+		mediaQuery.addEventListener("change", updater);
+
+		return () => mediaQuery.removeEventListener("change", updater);
```

Comment:
Can't we simplify that with the usage of `onOff()`?

```suggestion
		const mediaQuery = window.matchMedia(query);
		setActive(mediaQuery.matches);
		
		return onOff(mediaQuery, "change", (event: MediaQueryListEvent) => setActive(event.matches));
```

Reply:
We will lose support for Safari 13 with this implementation right now (as we have no fallback for the old implementation), which hit us in one of the projects earlier.

JFYI

Reply:
Wouldn't it be better to add support to `onOff()` instead of fixing it just in this one place?

Reply:
Adding support for `onOff` for `addListener`? That would be weird, as `onOff` is for events, and `addListener` is MediaQuery-specific.